### DISCUSSION TITLE
Scaling down logic

### DISCUSSION THREAD

**Discussion by Tazovsky** [kedacore/keda#3551] [Q&A / Need Help]
I am using Keda Metrics API scaler. My API checks which pods have user logged into pods, sends info to Keda, Keda scales up if there is no free pod - in my setup each user gets pods exclusively. But when API sends metric value to scale down, then Keda may kill pods which have active user logged in. 

For example, et's assume I have 5 pods:

1. have user logged in
2. have user logged in
3. have user logged in
4. pod is free
5. pod is free

`targetValue == 1` and  value returned from API is `4` - Keda starts scaling down, so which pods will be terminated?

I also have following questions:

1. what is the Keda logic of scaling pods down - I mean how does Keda know which pod to terminate?
2. is there any way to tell Keda to terminate pods only with some specific property, tag, etc.?

**Comment by tomkerkhove** [comment]
KEDA is building on top of the built-in HPA of Kubernetes and only provides the metrics you provide, so in the end HPA will decide what pod will be terminated which is random.

I would suggest, however, to open a new issue (if none already exists) to keep track of scale down logic to determine instance to be killed (ie by calling an API). However, this will not happen soon.

**Reply by tomkerkhove** [reply]
It's a valid scenario though

**Comment by Tazovsky** [comment]
@tomkerkhove thanks for you response. I will create separate feature request. Can you please provide

> I would suggest, however, to open a new issue (if none already exists) to keep track of scale down logic to determine instance to be killed (ie by calling an API). However, this will not happen soon.

Could you at least estimate the delivery of this task - would it be a month, half a year, year or longer? That info would be very useful to redesign project roadmap, so thank you in advance! :)

**Reply by tomkerkhove** [reply]
Unfortunately I cannot given the dependency that we have and depends on other priorities; sorry.

**Reply by milad2golnia** [reply]
@tomkerkhove Can you please gives us an update about this feature? Is it implemented or it is still pending?

**Comment by milad2golnia** [comment]
@Tazovsky Did they implemented this feature? Can you please share *feature request* link for others to keep track of it?

**Comment by ptrajkumar** [comment]
If you are running a deployment , you can simply use controller.kubernetes.io/pod-deletion-cost to control scaling down.  Simply mark the pods with user logged in as higher cost
